### PR TITLE
Add tests for simulating external API calls, Fixes #302

### LINKED ISSUE #302: Add pytests that simulate external API calls
**I'm submitting a ...**
- [ ] bug report
- [ ] feature request
- [X] test request

**Current behavior:**

External requests like https://query-server.herokuapp.com/api/v1/search/gigablast are not currently tested.
**Expected behavior:**

Add tests to the Travis CI process that make external API calls to both valid and invalid scraper names.
**Steps to reproduce:**

### PR DESCRIPTION
**Pull request by shaddygarg**
Fixes #302 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Added tests for simulating external API calls
-

### CODE REVIEW

**Comment by shaddygarg** [comment]
@cclauss The travis tests are passing for python 2.7. The app is not deployable on python 3.6 due to issues with relative imports ([Travis log](https://travis-ci.org/fossasia/query-server/jobs/297358431#L560)) being frequently used in the project. How should I proceed further?

**Comment by cclauss** [comment]
Awesome work!  Thanks a ton for doing this.

You can try marking your tests as [expect to fail on Python 3](https://github.com/fossasia/query-server/pull/297/files#diff-eff0198ff37b58d7351b62ca25b3f095R46) by:
1. import pytest  # at the top of the file with other imports
2. @pytest.mark.xfail(sys.version_info >= (3, 0), reason='Python 3 blocked until #297 goes live')
    * That must go on the line just before  the __def__ of every function that you expect to fail.

File: app/test_server.py [hunk 1]
```diff
@@ -7,7 +7,63 @@
...
 def test_true():
     assert True, "We have a problem!"
+
+
+def test_invalid_url_api_call():
+    response = requests.get('http://localhost:7001/api/v1/search/invalid_url')
+    assert response.json()['Status Code'] == 404
+
+
+def test_google_api_call():
```
Comment (cclauss) [inline app/test_server.py]: engines = """ask ubaidu bing duckduckgo exalead google quora yahoo yandex
             tyoutube"""


def make_engine_api_call(engine_name):
    url = 'http://localhost:7001/api/v1/search/' + engine_name
    assert requests.get(url).json()['Status Code'] == 400, engine_name


def test_engine_api_calls(engine_names=None):
    for engine_name in (engine_names or engines.lower().split()):
        make_engine_api_call(engine_name)

**Comment by cclauss** [comment]
You might have to pytest.mark.skipif https://docs.pytest.org/en/latest/skipping.html

**Comment by shaddygarg** [comment]
@cclauss I have made changes to the .travis.yml file so that it runs differently for different versions of python.

**Comment by cclauss** [comment]
OK but please help me to move it back to the old way once we get relative imports checked in.

**Comment by shaddygarg** [comment]
Sure, will do it as soon as the relative imports are checked in!

**Comment by shaddygarg** [comment]
@cclauss Can you review the PR?

File: app/test_server.py [hunk 2]
```diff
@@ -7,7 +7,33 @@
...
+
+@pytest.mark.xfail(sys.version_info >= (3, 0),
+                   reason='Python 3 blocked until #297 goes live')
+def test_invalid_url_api_call():
+    response = requests.get('http://localhost:7001/api/v1/search/invalid_url')
+    assert response.json()['Status Code'] == 404
+
+
+@pytest.mark.xfail(sys.version_info >= (3, 0),
+                   reason='Python 3 blocked until #297 goes live')
```
Comment (cclauss) [inline app/test_server.py]: Lines 26 and 27 are not really required but I approve nevertheless.  Nice work!

**Comment by cclauss** [comment]
@bhaveshAn @niranjan94 could we please get your reviews on this one?

**Comment by shaddygarg** [comment]
@ParthS007 @bhaveshAn @niranjan94 Please review the PR.

**Pull request merged**